node/formats/eszip.ts

```diff
@@ -51,7 +52,10 @@ const bundleESZIP = async ({
 
   try {
     await deno.run(['run', ...flags, bundler, JSON.stringify(payload)], { pipeOutput: true })
-  } catch (error: unknown) {
+  } catch (_error: unknown) {
+    let error = _error
+    error = checkNpmImportError(error) ?? error
```

Comment:
I find this sequence confusing, to be honest.

I think a method called `checkNpmImportError` that returns either an error or `undefined` isn't the most natural?

I would either:

- Call it `isNpmImportError` and make it return a boolean — you'd then need to call the `NPMImportError` separately if that condition is true, or
- Call it something like `wrapNpmImportError` which always returns an error — either the original error if it's not an npm import error, or an instance of `NPMImportError` if it is

What do you think?

Reply:
Agreed! looks like I write too much go :D changed in https://github.com/netlify/edge-bundler/pull/195/commits/67475ddfd2de1e72bc69bca48e7545eed3defd0e